### PR TITLE
feat: add option to provide extra schemas

### LINKED ISSUE #9: support yaml
I'm using `.clang-format` file, schemas provided by [yamlls](https://github.com/redhat-developer/yaml-language-server), and I have to add schemas manually.

Is it possible to support yaml?

Thans a lot!

### PR DESCRIPTION
**Pull request by williamboman**
I also took the liberty to format some of the code according to the
Stylua formatting rules, hope you don't mind! I also added EmmyLua
type definitions for the `opts` table.

Supersedes #13.

Closes #9.

### CODE REVIEW

**Review by b0o — CHANGES_REQUESTED** [review]
William, thank you for your contribution, and I apologize for how long it's taken me to review. 

I've left a few review comments, if you have time, please look them over.

File: README.md [hunk 1]
```diff
@@ -106,28 +106,41 @@ If you want to use your own schemas in addition to schemas from SchemaStore, you
...
+require('lspconfig').jsonls.setup {
+  settings = {
+    json = {
+      schemas = require('schemastore').yaml.schemas()
+      validate = true,
+    },
+  },
+}
+```
+
```
[the remark this answers is not in the document]
Comment (williamboman) [reply README.md:143]: Yeah for sure! I'm actually not sure why I added it in the first place (also with an incorrect sample).

[the remark this answers is not in the document]
Comment (williamboman) [reply README.md]: Reverted in c2c1ba9e58a1c8c4e8f0dd39a8ab38321dc1a501.

File: test/schemastore.lua [hunk 2]
```diff
@@ -171,6 +171,38 @@ Describe('the schemastore.init module', function()
...
+        -- Should not mutate the original extra table
+        Expect(extra).To.Be.DeepEqual {
+          {
+            description = 'My Custom JSON schema',
+            fileMatch = { 'foobar.json', '.foobar.json' },
+            name = 'foobar.json',
+            url = 'https://example.com/schema/foobar.json',
+          },
+        }
+      end)
```
[the remark this answers is not in the document]
Comment (williamboman) [reply test/schemastore.lua:227]: Added tests in e2d792ae9d885f6797313e418d9236d0079e2f59. I might be missing something but `extra + replace` doesn't seem like it was properly supported currently so I had to make some implementation changes as well.

[the remark this answers is not in the document]
Comment (williamboman) [reply test/schemastore.lua:227]: Haha nvm I just realized I am the one who's adding `extra` as a new feature 😅. Anyways, wdyt about the proposed implementation change of extending the catalog as a basis for applying the extra schemas?

File: lua/schemastore/init.lua [hunk 3]
```diff
@@ -24,70 +24,78 @@ function M.json.get(name)
...
+    end)
     for _, index in ipairs(ignore) do
       table.remove(schemas, index)
     end
   end
 
-  return schemas
+  if type(opts.extra) == 'table' and not vim.tbl_isempty(opts.extra) then
+    -- We { unpack(opts.extra) } as to not mutate opts.extra
+    return vim.list_extend({ unpack(opts.extra) }, schemas)
```
[the remark this answers is not in the document]
Comment (williamboman) [reply lua/schemastore/init.lua]: :+1: In my view these extra schemas should simply override the bundled ones (basically `replace`)? Or should it maybe produce an error?

**Comment by b0o** [comment]
Looks great, thank you William!

**Review by b0o — APPROVED** [review]

**Pull request merged**